### PR TITLE
Const generics

### PR DESCRIPTION
**Pull request by cwlittle**

### CODE REVIEW

**Comment by cwlittle** [comment]
Looks like the coverage changes are mostly misfires. 242 is a closing bracket of a match self block, 427-429 are the #[inline] tag, function definition, and opening line of a for loop. 431 is the closing bracket of that same for loop.

**Review by mappum — COMMENTED** [review]
Dope 💯 Left a few comments and I see there are some warnings but can merge after dealing with that.

**Comment by cwlittle** [comment]
@mappum Just now seeing these comments. Getting to them now.

File: src/lib.rs [hunk 1]
```diff
@@ -378,101 +378,49 @@ tuple_impl!(A, B, C, D; E);
...
-                    sum += element.encoding_length()?;
-                }
-                Ok(sum)
-            }
+impl<T: Encode + Terminated, const N: usize> Encode for [T; N] {
+    #[allow(non_snake_case, unused_mut, unused_variables)]
+    #[inline]
+    fn encode_into<W: Write>(&self, mut dest: &mut W) -> Result<()> {
+        for element in self[..].iter() {
+            element.encode_into(&mut dest)?;
```
[the remark this answers is not in the document]
Comment (cwlittle) [reply src/lib.rs:386]: If we remove the mutable borrow, the compiler complains that the dest parameter doesn't need to be mutable. Not entirely sure if we should suppress the warning or change the signature here. The change would be 
```
fn encode_into<W: Write>(&self, dest: &mut W) -> Result<()>
```

[the remark this answers is not in the document]
Comment (mappum) [reply src/lib.rs:397]: Changing the signature is preferred since the leading mut doesn't actually change the type, it's just the same as `let mut`.

**Pull request merged**